nerfstudio/process_data/equirect_utils.py

```
@@ -0,0 +1,120 @@
+        TextColumn("[bold blue]Generating Planar Images", justify="right"),
+        BarColumn(),
+        TaskProgressColumn(show_speed=True),
+        ItersPerSecColumn(suffix="equirect frames/s"),
+        TimeRemainingColumn(elapsed_when_finished=True, compact=True),
+    )
+
+    curr_im = 0
+    with progress:
+        for i in progress.track(os.listdir(frame_dir), description="", total=num_ims):
+            if i.lower().endswith((".jpg", ".png", ".jpeg")):
+                im = np.array(cv2.imread(os.path.join(frame_dir, i)))
+                im = torch.tensor(im, dtype=torch.float32, device=device)
+                im = torch.permute(im, (2, 0, 1)) / 255.0
+                count = 0
+                for u_deg, v_deg in yaw_pitch_pairs:
+                    v_rad = torch.pi * v_deg / 180.0
+                    u_rad = torch.pi * u_deg / 180.0
+                    pers_image = equi2pers(im, rots={"roll": 0, "pitch": v_rad, "yaw": u_rad}) * 255.0
+                    pers_image = (pers_image.permute(1, 2, 0)).type(torch.uint8).to("cpu").numpy()
+                    cv2.imwrite(f"{output_dir}/{i[:-4]}_{count}.jpg", pers_image)
+                    count += 1
+            curr_im += 1
+
+    return output_dir
+
+
+def compute_resolution_from_equirect(image_dir: Path, num_images: int) -> Tuple[int, int]:
+    """Compute the resolution of the persepctive projections of equirectangular images
+       from the heuristic: num_image * res**2 = orig_height * orig_width.
+
+    Args:
+        image_dir: The directory containing the equirectangular images.
+    returns:
+        The target resolution of the perspective projections.
+    """
+
+    for i in os.listdir(image_dir):
+        if i.lower().endswith((".jpg", ".png", ".jpeg")):
+            im = np.array(cv2.imread(os.path.join(image_dir, i)))
+            res_squared = (im.shape[0] * im.shape[1]) / num_images
+            return (int(np.sqrt(res_squared)), int(np.sqrt(res_squared)))
+    return (1200, 1200)
```

You should raise an error if no images are found.